server/graphql/v1/TransactionInterface.js

```diff
@@ -235,32 +290,34 @@ export const TransactionOrderType = new GraphQLObjectType({
       description: {
         type: GraphQLString,
         resolve(transaction) {
-          return transaction.description || transaction.getOrder().then(order => order && order.description);
+          const getOrder = transaction.getOrder
+            ? transaction.getOrder().then(order => order && order.description)
+            : null;
+          return transaction.description || getOrder;
         },
       },
       privateMessage: {
         type: GraphQLString,
         resolve(transaction) {
-          // TODO: Put behind a login check
-          return transaction.getOrder().then(order => order && order.privateMessage);
+          return transaction.getOrder ? transaction.getOrder().then(order => order && order.privateMessage) : null;
```

Comment:
Why do we add all this defensive code to check if methods are available?

Reply:
TransactionInterface is supposed to be expecting a Sequelize Model but when we parse the transactions coming from the ledger, those transactions are just object(with some fields like the Sequelize model) but they don't have their methods. That's why the validation is necessary.

Reply:
Can you explain me why it's ok to send `null` there when there is no method? Aren't we expecting a value?

Reply:
I've checked every `return null` case for those validations. I don't think we're waiting these specific values where we return null/undefined. But also in all other scenarios we have, we are using the sequelize models and the validations I added won't never be catched other than the `transactionsFromLedger`